providers/flagd/pkg/provider.go

```diff
@@ -235,147 +195,147 @@ func (p *Provider) setStatus(status of.State) {
 
 // ProviderOptions
 
-type ProviderOption func(*Provider)
+type ProviderOption func(*ProviderConfiguration)
```

Comment:
should those methods also be part of the configuration.go? as they are really configuration related?